src/transaction/rw_transaction.rs

```diff
@@ -152,22 +152,21 @@ impl<'db, 'txn> RwTransaction<'db> {
     ///     let rw = db.rw_transaction()?;
     ///
     ///     // Remove a value
-    ///     rw.remove(Data { id: 1 })?;
+    ///     let old_value = rw.remove(Data { id: 1 })?;
     ///
     ///     // /!\ Don't forget to commit the transaction
     ///     rw.commit()?;
     ///
     ///     Ok(())
     /// }
     /// ```
-    // TODO: Return the Option<T> of the removed value
-    pub fn remove<T: Input>(&self, item: T) -> Result<()> {
+    pub fn remove<T: Input>(&self, item: T) -> Result<DatabaseOutputValue> {
         let (watcher_request, binary_value) = self
             .internal
             .concrete_remove(T::native_db_model(), item.to_item())?;
-        let event = Event::new_delete(binary_value);
+        let event = Event::new_delete(binary_value.clone());
         self.batch.borrow_mut().add(watcher_request, event);
-        Ok(())
+        Ok(binary_value)
```

Comment:
Why not return `T` ?

Reply:
I wanted to change as little of the existing tests as possible and minimize the diff